services/transactionService.ts

```diff
@@ -163,45 +111,26 @@ export async function upsertManyTransactionsForAddress (transactions: Transactio
   return ret.filter((t) => t !== undefined) as TransactionWithAddressAndPrices[]
 }
 
-export async function syncTransactionsForAddress (addressString: string): Promise<TransactionWithAddressAndPrices[]> {
-  const address = await fetchAddressBySubstring(addressString)
-  let newTransactionsCount = -1
-  let seenTransactionsCount = 0
-  let insertedTransactions: TransactionWithAddressAndPrices[] = []
-  while (newTransactionsCount !== 0) {
-    const addressTransactions = await getAddress({
-      address: address.address,
-      nbFetch: FETCH_N,
-      nbSkip: seenTransactionsCount
-    })
-    let nextConfirmedTransactions = addressTransactions.confirmedTransactionsList
-    let nextUnconfirmedTransactions = addressTransactions.unconfirmedTransactionsList.map(mempoolTx => parseMempoolTx(mempoolTx))
-
-    // remove transactions older than the networks
-    nextConfirmedTransactions = nextConfirmedTransactions.filter(txThesholdFilter(address))
-    nextUnconfirmedTransactions = nextUnconfirmedTransactions.filter(txThesholdFilter(address))
+export async function syncAllTransactionsAndPricesForAddress (addressString: string, returnMaxTransactions: number): Promise<TransactionWithAddressAndPrices[]> {
+  addressString = parseAddress(addressString)
+  const parameters = {
+    addressString,
+    start: 0,
+    maxTransactions: returnMaxTransactions
+  }
 
-    newTransactionsCount = nextConfirmedTransactions.length // Unconfirmed transactions are NOT taken into account when skipping with nbSkip
-    seenTransactionsCount += newTransactionsCount
+  const insertedTransactions: TransactionWithAddressAndPrices[] = await syncTransactionsAndPricesForAddress(parameters)
 
-    let newInsertedTransactions = await upsertManyTransactionsForAddress(nextConfirmedTransactions, address)
-    newInsertedTransactions = [
-      ...newInsertedTransactions,
-      ...(await upsertManyTransactionsForAddress(nextUnconfirmedTransactions, address, false))
-    ]
-    insertedTransactions = [...insertedTransactions, ...newInsertedTransactions]
-    await new Promise(resolve => setTimeout(resolve, FETCH_DELAY))
+  if (parameters.maxTransactions === Infinity || insertedTransactions.filter(t => t.confirmed).length < parameters.maxTransactions) {
```

Comment:
I don't understand this  `< `. Why run the else clause if we already inserted more transactions than `parameters.maxTransactions`? Maybe add some comments clarifying.

Reply:
Because we always want to sync all transactions to our database, the `maxTransactions` is just how many transactions we are going to return now to the caller. For sure there is a better name to use here instead of `maxTransactions`...

Reply:
`maxTransactionsToReturn` is clearer, changed

Reply:
In the future we won't sync the full history when we have already synced once at least, we will stop as I proposed yesterday on #403